### ISSUE TITLE
skopeo container copy API version unsupported by newer docker

### ISSUE THREAD

**Issue by mcdonc** [cachix/devenv#955]
With docker-ce installed as per https://www.digitalocean.com/community/tutorials/how-to-install-and-use-docker-on-ubuntu-22-04 , trying to generate and run a container image produces the following error.

```
DEBU[0001] Walking filesystem: /nix/store/7j3kv2hh983js6vh01nzq48qg5r7i3ai-entrypoint 
DEBU[0001] No compression detected                      
DEBU[0001] Using original blob without modification     
Copying blob 6a43ce11f7a7 [===================================>] 1.5GiB / 1.5GiB
DEBU[0010] ... streaming done                           
DEBU[0010] Sending as tar file 6a43ce11f7a783fe039d1cf01ceacf79706912dcade4671c0Copying blob 6a43ce11f7a7 done  
DEBU[0010] docker-daemon: sending done, status saving image to docker engine: Error response from daemon: client version 1.22 is too old. Minimum supported API Copying blob 6a43ce11f7a7 done  
DEBU[0010] docker-daemon: Closing tar stream to abort loading 
FATA[0010] writing blob: saving image to docker engine: Error response from daemon: 
client version 1.22 is too old. Minimum supported API version is 1.24, please upgrade your client to a newer version 
```

On Ubuntu 22.04, but with the upstream docker-ce apt repository.  Here's the dockerd version:

```
chrism@ubuntu-22:~/projects/devenv$ dockerd -v
Docker version 25.0.1, build 71fa3ab
```

**Comment by mcdonc** [comment]
A workaround is to install an older version of docker CE ala https://askubuntu.com/questions/1213841/how-to-install-specific-docker-version-from-apt-cache-policy-docker-ce-candidate

Docker version 24.0.8 seems to work for me.

**Comment by mcdonc** [comment]
Upstream issue at https://github.com/containers/skopeo/issues/2202

**Comment by mcdonc** [comment]
This will be fixed via https://github.com/cachix/devenv/pull/963

**Comment by domenkozar** [comment]
Released as skopeo 1.14.2 and part of nixpkgs now
